### PR TITLE
Improved compatibility with code coverage.

### LINKED ISSUE #190: Bring compatibility with code coverage
Apparently, Istanbul (even the latest one) doesn't support some ES6 constructs that we use.  This means that we're unable to run jobs, because they take ridiculous amount of time.

There are two things we need to do:

1. Fork benderjs-coverage to use a standard Istanbul (benderjs-coverage uses a hacked one, because, AFAIR, CKE4 requires that) and in its latest version.
2. Check what ES6 features break Istanbul and stop using them. My guess will be that deconstructuring in default params.

See https://github.com/benderjs/benderjs/issues/253 for more.

### PR DESCRIPTION
**Pull request by oskarwrobel**
Fixes #190 

Related: https://github.com/ckeditor/ckeditor5-engine/pull/497

### CODE REVIEW

File: dev/tests/build/utils.js [hunk 1]
```diff
@@ -188,20 +188,6 @@ describe( 'build-utils', () => {
 		} );
 	} );
 
-	describe( 'getBabelPlugins', () => {
```
Comment (Reinmar) [inline dev/tests/build/utils.js]: Why removing a correct test?

Comment (oskarwrobel) [inline dev/tests/build/utils.js]: Was duplicated.

**Comment by Reinmar** [comment]
I pushed a commit upgrading also other Babel plugins, because there was incompatibility between packages I had.

**Comment by Reinmar** [comment]
I noticed that with this set of Babel plugins it also transpiles arrow functions :( That's sad. Can we avoid that?

**Comment by Reinmar** [comment]
To confirm - check transpiled version of `utils/locale.js` and `this` that appeared in the constructor.

**Comment by Reinmar** [comment]
Hm... on the other hand – that doesn't happen in .e.g. `undo/undoengine.js`  in the `init()` function.

**Comment by Reinmar** [comment]
screen shot 2016-06-20 at 14 39 26

:(

**Comment by Reinmar** [comment]
This is due to the destructuring transpilation. If we disable it then default params with destructuring still don't work.

**Comment by Reinmar** [comment]
https://github.com/gotwarlost/istanbul/issues/613

**Comment by Reinmar** [comment]
https://github.com/estools/escodegen/issues/296 has a PR, but escodegen looks to be quite dead ;/. Things don't look super well.

I'm afraid we'll need to just get rid of default params from our code :( Though, it's worth checking if we can't use default params with destructuring or default params in general or destructuring in general.

**Comment by oskarwrobel** [comment]
I also think that we should avoid transforming code in build task as long as possible. Babel might inject some inline helpers to each file what could increase size of a bundle. We can try to use https://babeljs.io/docs/plugins/external-helpers/ but it makes everything more complicated.

**Pull request merged**